src/v/serde/parquet/tests/generate_file.cc

```diff
@@ -237,15 +245,83 @@ schema_element all_types_schema() {
         leaf_node("C", field_repetition_type::required, i64_type{}),
         leaf_node("D", field_repetition_type::required, f32_type{}),
         leaf_node("E", field_repetition_type::required, f64_type{}),
-        leaf_node("F", field_repetition_type::required, byte_array_type{})),
-      group_node("Logical", field_repetition_type::optional));
+        leaf_node("F", field_repetition_type::required, byte_array_type{})));
+    // TODO: also add logical types
     index_schema(document_schema);
     return document_schema;
 }
 
+value generate_value(const schema_element& root);
+
+value generate_group(const schema_element& root) {
+    group_value g;
+    g.reserve(root.children.size());
+    for (const auto& child : root.children) {
+        g.emplace_back(generate_value(child));
+    }
+    return g;
+}
+
+value generate_required(const schema_element& root) {
+    return ss::visit(
+      root.type,
+      [&root](const std::monostate&) { return generate_group(root); },
+      [](const bool_type&) -> value {
+          return boolean_value{random_generators::get_real<float>() < 0.5};
+      },
+      [](const i32_type&) -> value {
+          return int32_value{random_generators::get_int<int32_t>()};
+      },
+      [](const i64_type&) -> value {
+          return int64_value{random_generators::get_int<int64_t>()};
+      },
+      [](const f32_type) -> value {
+          return float32_value{random_generators::get_real<float>()};
+      },
+      [](const f64_type&) -> value {
+          return float64_value{random_generators::get_real<double>()};
+      },
+      [](const byte_array_type& t) -> value {
+          if (t.fixed_length) {
+              return fixed_byte_array_value{iobuf::from(
+                random_generators::gen_alphanum_string(*t.fixed_length))};
+          }
+          auto size = random_generators::get_int<size_t>(64);
+          return byte_array_value{
+            iobuf::from(random_generators::gen_alphanum_string(size))};
+      });
+}
+
+value generate_optional(const schema_element& root) {
+    if (random_generators::get_real<float>() > 0.82) {
+        return null_value();
+    }
+    return generate_required(root);
+}
+
+value generate_repeated(const schema_element& root) {
+    repeated_value r;
+    size_t n = 0;
+    r.reserve(n);
```

Comment:
n = 0?

Reply:
Haha thanks - updated to use a random number